### PR TITLE
Integration vb

### PR DESCRIPTION
**Pull request by sczembor**
+ integration tests for vb
+ move `instantiate_congress` to common

### CODE REVIEW

File: voting_body/tests/integration.rs [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+#[serde(crate = "near_sdk::serde")]
+pub struct TokenMetadata {
+    pub class: u64,
+    pub issued_at: Option<u64>,
+    pub expires_at: Option<u64>,
+    pub reference: Option<String>,
+    pub reference_hash: Option<String>,
+}
+
+pub struct InitStruct {
```
Comment (robert-zaremba) [inline voting_body/tests/integration.rs:26]:
```suggestion
pub struct Suite {
```

File: voting_body/tests/integration.rs [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+        let res = user.call(registry.id(), "is_human_call_lock")
+        .args_json(json!({"ctr": voting_body.id(), "function": "vote", "payload": serde_json::to_string(payload).unwrap(), "lock_duration": 1800000, "with_proof": false}))
+        .max_gas()
+        .deposit(parse_near!("1 N"))
+        .transact()
+        .await?;
+        assert!(res.is_success(), "{:?}", res);
+    }
+    Ok(())
+}
```
Comment (robert-zaremba) [inline voting_body/tests/integration.rs:54]: let's do it as a method of the `Suite`

File: voting_body/tests/integration.rs [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        john,
+        admin,
+        proposal_id,
+        registry_contract,
+        vb_members: vec![vb_member1, vb_member2, vb_member3, vb_member4],
+    })
+}
+
+#[tokio::test]
+async fn veto() -> anyhow::Result<()> {
```
Comment (robert-zaremba) [inline voting_body/tests/integration.rs:192]: we should test 2 veto: one by VB and other by CoA

File: voting_body/tests/integration.rs [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+        .await?
+        .json()?;
+
+    assert_eq!(ProposalStatus::Vetoed, proposal.unwrap().proposal.status);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn dismiss() -> anyhow::Result<()> {
```
Comment (robert-zaremba) [inline voting_body/tests/integration.rs:265]: ideally we need test for `DismissAndBan` proposal

File: voting_body/tests/integration.rs (hunk 3 above)
Comment (sczembor) [reply voting_body/tests/integration.rs:192]: the CoA veto test is already in congress integration test

File: voting_body/tests/integration.rs (hunk 4 above)
Comment (sczembor) [reply voting_body/tests/integration.rs:265]: this test is already in congress integration test, VB doesn't have this power

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**